backend/x11/backend.c

```diff
@@ -188,6 +188,24 @@ static void init_atom(struct wlr_x11_backend *x11, struct wlr_x11_atom *atom,
 	atom->reply = xcb_intern_atom_reply(x11->xcb_conn, atom->cookie, NULL);
 }
 
+
+static void parse_xcb_setup(struct wlr_output *output, xcb_connection_t *xcb_conn) {
+	const xcb_setup_t *xcb_setup;
+	unsigned int length;
+
+	xcb_setup = xcb_get_setup(xcb_conn);
+	length = xcb_setup_vendor_length (xcb_setup);
+	if (length > sizeof(output->make)-1) {
+		length = sizeof(output->make)-1;
+	}
+	memcpy (output->make, xcb_setup_vendor (xcb_setup), length);
```

Comment:
I'd prefer that snprintf is used for this as well, which would actually remove the need for a lot of the preceding code:
```c
snprintf(output->make, sizeof(output->make), "%.*s",
    xcb_setup_vendor_length(xcb_setup),
    xcb_setup_vendor(xcb_setup));
```

Reply:
Why not strncpy?

Reply:
xcb_setup_vendor is not null necessarily null terminated.

Reply:
That's what the %.*s of printf is about, it's safe :)

Reply:
@martinetd but it could still outgrow output->make

Reply:
@martinetd ...no it can't sorry for the noise.

Reply:
That's why it's `s*n*printf` the output->make size is the second argument to the function.

The maximum string length is then given as format option.